### PR TITLE
fix bug: remove tail whitespace from gt label

### PR DESCRIPTION
**Pull request by cuhk-hbsun**

### CODE REVIEW

File: mmocr/utils/string_util.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            the string. Default: True.
+        strip_pos (str): Which position for removing, can be one of
+            ('both', 'left', 'right'), Default: 'both'.
+        strip_str (str|None): A string specifying the set of characters
+            to be removed from the left and right part of the string.
+            If None, all leading and trailing whitespaces
+            are removed from the string. Default: None.
+    """
+
+    def __init__(self, strip=True, strip_pos='both', strip_str=None, **kwargs):
```
Comment (innerlee) [inline mmocr/utils/string_util.py:16]: remove unused arg `kwargs`

File: mmocr/datasets/utils/parser.py [hunk 2]
```diff
@@ -53,10 +57,13 @@ def __init__(self, keys=[], **kwargs):
         assert isinstance(keys, list)
         assert len(keys) > 0
         self.keys = keys
+        self.strip_cls = StringStrip(**kwargs)
 
     def get_item(self, data_ret, index):
         map_index = index % len(data_ret)
-        line_json_obj = json.loads(data_ret[map_index])
+        json_str = data_ret[map_index]
+        json_str = self.strip_cls(json_str)
```
Comment (gaotongxiao) [inline mmocr/datasets/utils/parser.py:65]: Is it necessary to strip raw json string? `json.loads` can handle that.

File: mmocr/utils/string_util.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        return in_str
+
+    def _rstrip(self, in_str):
+        if self.strip_str is None:
+            in_str = in_str.rstrip()
+        else:
+            in_str = in_str.rstrip(self.strip_str)
+
+        return in_str
```
Comment (gaotongxiao) [inline mmocr/utils/string_util.py:47]: Uncessary wrappers? 
Take `_strip()` as an example, `in_str = in_str.strip(self.strip_str)` should be fine for every possible values including `None`.

File: mmocr/utils/string_util.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    def __call__(self, in_str):
+
+        if not self.strip:
+            return in_str
+
+        if self.strip_pos == 'left':
+            return self._lstrip(in_str)
```
Comment (gaotongxiao) [inline mmocr/utils/string_util.py:46]: `return in_str.lstrip(self.strip_str)`

File: mmocr/utils/string_util.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    def _lstrip(self, in_str):
+        in_str = in_str.lstrip(self.strip_str)
+
+        return in_str
+
+    def _rstrip(self, in_str):
+        in_str = in_str.rstrip(self.strip_str)
+
+        return in_str
```
Comment (gaotongxiao) [inline mmocr/utils/string_util.py:38]: del

File: mmocr/utils/string_util.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+        if not self.strip:
+            return in_str
+
+        if self.strip_pos == 'left':
+            return self._lstrip(in_str)
+        elif self.strip_pos == 'right':
+            return self._rstrip(in_str)
+        else:
+            return self._strip(in_str)
```
Comment (gaotongxiao) [inline mmocr/utils/string_util.py:50]: similar as above

**Comment by gaotongxiao** [comment]
For `LineJsonParser`, we may need to apply the stripping operation on each field *after* `json.loads` have parsed the raw json string.

**Comment by gaotongxiao** [comment]
Also `LineStrParser` should follow the same logic. That is, apply `strip_cls` to each field after splitting.

**Review by gaotongxiao — APPROVED** [review]

**Pull request merged**